src/main/java/edu/harvard/dbmi/avillach/dictionaryetl/columnmeta/AbstractColumnMetaProcessor.java

```diff
@@ -0,0 +1,297 @@
+     */
+    public Optional<ColumnMeta> processColumnMetas(List<ColumnMeta> columnMetas) {
+        ColumnMeta columnMeta = null;
+        try {
+            if (columnMetas.size() == 1) {
+                columnMeta = columnMetas.getFirst();
+            } else {
+                boolean isContinuous = columnMetas.stream().anyMatch(meta -> !meta.categorical());
+                if (!isContinuous) {
+                    columnMeta = flattenCategoricalColumnMeta(columnMetas);
+                } else {
+                    columnMeta = flattenContinuousColumnMeta(columnMetas);
+                }
+            }
+        } catch (Exception e) {
+            log.error("Error processing concept path: {} with values for column metas: {}",
+                    columnMetas.getFirst().name(),
+                    e.getMessage());
+            columnMetaErrors.add(columnMetas);
+        } finally {
+            // Decrement the task counter no matter what happens
+            task.getAndDecrement();
+        }
+
+        return columnMeta != null ? Optional.of(columnMeta) : Optional.empty();
+    }
+
+    /**
+     * In some cases we need to flatten a group of ColumnMetaRows down to a single ColumnMeta.
+     * This is the case for Continuous Concept Paths in the columnMeta.csv that have more than one row.
+     *
+     * @param columnMetas A List of ColumnMeta where the first ColumnMeta is expected to be a continuous value.
+     * @return ColumnMeta that has a min and max based on all ColumnMetas in the list.
+     */
+    protected ColumnMeta flattenContinuousColumnMeta(List<ColumnMeta> columnMetas) {
+        // This is a special case. Where the parent (first element) being rolled into must be continuous.
+        if (columnMetas.getFirst().categorical()) {
+            throw new RuntimeException("Cannot flatten rows. Mixed concept types. Must be continuous OR categorical " +
+                                       "for a concept path.");
+        }
+
+        // As the list is processed the min and max will adjust to based on the "values" of other concepts.
+        final Double[] min = {columnMetas.getFirst().min()};
+        final Double[] max = {columnMetas.getLast().max()};
+
+        columnMetas.forEach(columnMeta -> {
+            if (columnMeta.categorical()) {
+                if (columnMeta.categoryValues().size() > 1) {
+                    throw new RuntimeException("Cannot flatten rows. Mixed concept types. Must be continuous OR categorical " +
+                                               "for a concept path.");
+                }
+
+                double value = Double.parseDouble(columnMeta.categoryValues().getFirst());
+                min[0] = Math.min(min[0], value);
+                max[0] = Math.max(max[0], value);
+            } else {
+                min[0] = Math.min(min[0], columnMeta.min());
+                max[0] = Math.max(max[0], columnMeta.max());
+            }
+        });
```

Comment:
Why are we using arrays here too? It feels like you could use just Double values for min and max?

Reply:
I will do the same thing as the above comment.